src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -252,4 +821,232 @@ public Request parse(String commandLine) {
+           authorization.setMechanism(Mechanism.DIGEST);
+           break;
+       default:
+           break;
+       }
+   }
+
+   /**
+    *
+    * Set the parameters of proxy server in http request advanced
+    *
+    * @param request         http request
+    * @param proxyServerParameters the parameters of proxy server
+    *
+    */
+   private void setProxyServer(Request request, String proxyServerParameters) {
+       if (!proxyServerParameters.contains("://")) {
+           proxyServerParameters = "http://" + proxyServerParameters;
+       }
+       URI uriProxy = null;
+       try {
+           uriProxy = new URI(proxyServerParameters);
+           request.setProxyServer("scheme", uriProxy.getScheme());
+           Optional<String> userInfoOptional = Optional.ofNullable(uriProxy.getUserInfo());
+           if (userInfoOptional.isPresent()) {
+               String userinfo = userInfoOptional.get();
+               if (userinfo.contains(":")) {
+                   String[] userInfo = userinfo.split(":");
+                   request.setProxyServer("username", userInfo[0]);
+                   request.setProxyServer("password", userInfo[1]);
+               }
+           }
+           Optional<String> hostOptional = Optional.ofNullable(uriProxy.getHost());
+           if (hostOptional.isPresent()) {
+               request.setProxyServer("servername", hostOptional.get());
+           }
+           if (uriProxy.getPort() != -1) {
+               request.setProxyServer("port", String.valueOf(uriProxy.getPort()));
+           } else {
+               request.setProxyServer("port", "1080");
+           }
+       } catch (URISyntaxException e) {
+           LOGGER.error("string '{}' cannot be converted to a URL", proxyServerParameters);
+           throw new IllegalArgumentException(proxyServerParameters + " cannot be converted to a URL");
+       }
+   }
+
+   /**
+    * Set the username and password of proxy server
+    *
+    * @param request               http request
+    * @param proxyServerUserPasswd the username and password of proxy server
+    */
+   private void setProxyServerUserInfo(Request request, String proxyServerUserPasswd) {
+       if (proxyServerUserPasswd.contains(":")) {
+           String[] userInfo = proxyServerUserPasswd.split(":");
+           request.setProxyServer("username", userInfo[0]);
+           request.setProxyServer("password", userInfo[1]);
+       }
+   }
+
+   /**
+    * Get post data by different type of data option
+    *
+    * @param postdata       the post data
+    * @param dataOptionName the different option of "--data"
+    * @return the post data
+    */
+   private String getPostDataByDifferentOption(String postdata, String dataOptionName) {
+       if (dataOptionName.equals("data-urlencode")) {
+           postdata = encodePostdata(postdata);
+       } else {
+           if (postdata.contains("@") && !dataOptionName.equals("data-raw")) {
+               postdata = postdata.replace("@", "");
```

Comment:
Is this really supposed to replace all `@` in the text? I thought the `@` has to be the first non space char.

Reply:
replace the @ if for getting the filename

Reply:
I think "Replace" will replace the first eligible character.

Reply:
I have change it to replacefirst()

Reply:
Well, my concern was, that the curl docs specify `--data-binary ARG` and state that: 

> If you start the data with the letter @, the rest should be a filename.

So I think the logic should not look for an `@` anywhere in the string, but specifically to be at the front of the string. I think it is safe to `trim` the arg before looking for the char.